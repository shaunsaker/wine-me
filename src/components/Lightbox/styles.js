import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  wrapper: {
    ...StyleSheet.absoluteFillObject,
    backgroundColor: styleConstants.colors.transBlack,
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    margin: styleConstants.dimensions.spacing.horizontal * 2,
  },
  card: {
    alignSelf: 'stretch',
    maxWidth: 680,
    backgroundColor: styleConstants.colors.white,
    borderRadius: styleConstants.dimensions.borderRadius,
    overflow: 'hidden',
  },
  headerContainer: {
    paddingVertical: styleConstants.dimensions.spacing.vertical,
    backgroundColor: styleConstants.colors.primary,
    ...styleConstants.shadows.small,
    justifyContent: 'center',
    alignItems: 'center',
  },
  titleText: {
    ...styleConstants.fonts.types.heading,
    color: styleConstants.colors.white,
  },
  iconContainer: {
    position: 'absolute',
    padding: styleConstants.dimensions.spacing.horizontal / 2,
    right: 0,
  },
  icon: {
    fontSize: styleConstants.fonts.sizes.icon,
    color: styleConstants.colors.white,
  },
});

export default styles;
