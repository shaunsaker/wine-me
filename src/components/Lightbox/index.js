import React from 'react';
import PropTypes from 'prop-types';
import { Actions } from 'react-native-router-flux';
import Animator from 'react-native-simple-animators';
import { View, Text } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styleConstants from '../../styleConstants';
import styles from './styles';

import Touchable from '../Touchable';

export default class Lightbox extends React.Component {
  constructor(props) {
    super(props);

    this.animateOut = this.animateOut.bind(this);
    this.onBack = this.onBack.bind(this);

    this.state = {
      shouldAnimateOut: false,
    };
  }

  static propTypes = {
    title: PropTypes.string,
    children: PropTypes.node,
    handleClose: PropTypes.func,
    disableClose: PropTypes.bool,
  };

  static defaultProps = {};

  animateOut() {
    this.setState({
      shouldAnimateOut: true,
    });
  }

  onBack() {
    const { handleClose } = this.props;

    if (handleClose) {
      handleClose();
    } else {
      Actions.pop();
    }
  }

  render() {
    const { shouldAnimateOut } = this.state;
    const { title, children, disableClose } = this.props;

    const closeComponent = !disableClose ? (
      <Touchable
        onPress={this.animateOut}
        style={styles.iconContainer}
        testID="lightbox.button.close"
      >
        <Icon name="close" style={styles.icon} />
      </Touchable>
    ) : null;

    return (
      <Animator
        type="translateY"
        initialValue={styleConstants.dimensions.window.height}
        finalValue={0}
        shouldAnimateIn
        shouldAnimateOut={shouldAnimateOut}
        animateOutCallback={this.onBack}
        style={styles.wrapper}
      >
        <View style={styles.container}>
          <View style={styles.card}>
            <View style={styles.headerContainer}>
              {closeComponent}

              <Text style={styles.titleText}>{title}</Text>
            </View>
            {children}
          </View>
        </View>
      </Animator>
    );
  }
}
