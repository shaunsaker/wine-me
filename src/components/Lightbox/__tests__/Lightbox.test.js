import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import Lightbox from '..';

// Fixes _bezier is not a function bug
jest.useFakeTimers();

describe('Lightbox', () => {
  const spies = [];
  const title = 'Test';
  const handleClose = jest.fn();
  const children = <View />;

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<Lightbox />);

      expect(component).toMatchSnapshot();
    });

    it('renders with all props', () => {
      const component = renderer.create(
        <Lightbox title={title} handleClose={handleClose}>
          {children}
        </Lightbox>,
      );

      expect(component).toMatchSnapshot();
    });

    it('renders disableClose state', () => {
      const component = renderer.create(
        <Lightbox title={title} handleClose={handleClose} disableClose>
          {children}
        </Lightbox>,
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    it('should handle animateOut', () => {
      const component = renderer.create(
        <Lightbox title={title} handleClose={handleClose}>
          {children}
        </Lightbox>,
      );
      const instance = component.getInstance();

      instance.animateOut();

      expect(instance.state.shouldAnimateOut).toBe(true);
    });

    describe('should handle onBack', () => {
      it('when handleClose is provided', () => {
        const component = renderer.create(
          <Lightbox title={title} handleClose={handleClose}>
            {children}
          </Lightbox>,
        );
        const instance = component.getInstance();

        instance.onBack();

        expect(handleClose).toHaveBeenCalled();
      });

      it('when handleClose is not provided', () => {
        const component = renderer.create(<Lightbox title={title}>{children}</Lightbox>);
        const instance = component.getInstance();

        instance.onBack();

        expect(handleClose).not.toHaveBeenCalled();
      });

      afterEach(() => {
        handleClose.mockClear();
      });
    });
  });

  describe('actions', () => {
    spies[0] = jest.spyOn(Lightbox.prototype, 'animateOut');
    const component = renderer.create(<Lightbox title={title}>{children}</Lightbox>);
    const { root } = component;
    const closeButton = root.findByProps({ testID: 'lightbox.button.close' });

    closeButton.props.onPress();

    expect(spies[0]).toHaveBeenCalled();
  });

  afterEach(() => {
    spies.forEach((spy) => {
      if (spy) {
        spy.mockClear();
      }
    });
  });
});
