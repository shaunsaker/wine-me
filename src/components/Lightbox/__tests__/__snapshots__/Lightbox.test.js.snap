// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lightbox renders renders disableClose state 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.70)",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": 1334,
        },
      ],
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "margin": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 5,
          "maxWidth": 680,
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#48466D",
            "borderColor": null,
            "borderWidth": 0,
            "elevation": 2,
            "justifyContent": "center",
            "paddingVertical": 10,
            "shadowColor": "#212121",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.17,
            "shadowRadius": 2,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Nunito-SemiBold",
              "fontSize": 17,
              "marginLeft": 15,
              "marginRight": 39.5,
            }
          }
        >
          Test
        </Text>
      </View>
      <View />
    </View>
  </View>
</View>
`;

exports[`Lightbox renders renders with all props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.70)",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": 1334,
        },
      ],
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "margin": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 5,
          "maxWidth": 680,
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#48466D",
            "borderColor": null,
            "borderWidth": 0,
            "elevation": 2,
            "justifyContent": "center",
            "paddingVertical": 10,
            "shadowColor": "#212121",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.17,
            "shadowRadius": 2,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Nunito-SemiBold",
              "fontSize": 17,
              "marginLeft": 15,
              "marginRight": 39.5,
            }
          }
        >
          Test
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "padding": 7.5,
              "position": "absolute",
              "right": 0,
            }
          }
          testID="lightbox.button.close"
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View />
    </View>
  </View>
</View>
`;

exports[`Lightbox renders renders with minimum required props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.70)",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": 1334,
        },
      ],
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "margin": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 5,
          "maxWidth": 680,
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#48466D",
            "borderColor": null,
            "borderWidth": 0,
            "elevation": 2,
            "justifyContent": "center",
            "paddingVertical": 10,
            "shadowColor": "#212121",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.17,
            "shadowRadius": 2,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Nunito-SemiBold",
              "fontSize": 17,
              "marginLeft": 15,
              "marginRight": 39.5,
            }
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "padding": 7.5,
              "position": "absolute",
              "right": 0,
            }
          }
          testID="lightbox.button.close"
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
